### ISSUE TITLE
Is `jax.scipy.stats.norm.logcdf` twice differentiable?

### ISSUE THREAD

**Issue by Cauch-BS** [jax-ml/jax#25564]
### Description

It seems that `jax.scipy.stats.norm.logcdf` is not twice differentiatiable. 

I implemented the following function in `jax`. 

```
@jax.jit
def log_likelihood_fn(
    log_lambda_x: Array,
    s_i: Array,
):
    # some more code here
    log_likelihoods = (
        jstat.norm.logcdf(
            -(
                log_rate / (s_i * jnp.sqrt(2))
            ),
            scale=1 / jnp.sqrt(2),
        )
    )

    return - log_likelihoods.sum()
   ```
   where `jstat.norm.logcdf(., scale = 1/jnp.sqrt(2))` is used in placed of `jax.scipy.special.erfc`, due to numerical instability problems with the latter. **s_i is very small, approximately in the scale of ~1e-5**. 
   
 I am trying to minimize this function (i.e. maximize log likelihood) using Newton's method. However, when computing the second derivative I found some wierd behavior. Drawing a simple graph with `matplotlib` for a single `lambda`, the graph appears to be convex. 

![image](https://github.com/user-attachments/assets/1aa9914a-3172-4e3d-9023-97d4c26e29fb)

However, drawing the second derivative, instead of being positive, as I expected (since it is a convex function), it appears that it oscillates wildly and blows up. 

![image](https://github.com/user-attachments/assets/db488204-982b-4741-af23-133b77832080)

I think that I should add a custom derivative to `jstat.norm.logcdf`, but I am unsure of how to get started. What causes this numerical instability in the twice-derivative calculation of `jstat.norm.logcdf`?  Any help would be appreciated. 

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.35
jaxlib: 0.4.34
numpy:  2.0.1
python: 3.12.4 | packaged by conda-forge | (main, Jun 17 2024, 10:23:07) [GCC 12.3.0]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='apricot', release='6.8.0-49-generic', version='#49-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov  4 02:06:24 UTC 2024', machine='x86_64')


$ nvidia-smi
Wed Dec 18 17:17:20 2024       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.54.15              Driver Version: 550.54.15      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA L4                      Off |   00000000:03:00.0 Off |                    0 |
| N/A   87C    P0             56W /   72W |   21264MiB /  23034MiB |    100%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
...
|    0   N/A  N/A   3109822      C   ...conda/envs/shape-reg/bin/python3.12        186MiB |
|    0   N/A  N/A   3114678      C   ...conda/envs/shape-reg/bin/python3.12        186MiB |
+-----------------------------------------------------------------------------------------+
```

**Comment by jakevdp** [comment]
`norm.logcdf` is implemented in terms of `special.log_ndtr`, which in turn is implemented using an asymptotic series. The second derivative plots you're showing look to me like the result of floating point rounding within that asymptotic series computation.

One question to help diagnose this: how do the results look if you enable float64 precision?

**Comment by Cauch-BS** [comment]
Enabling `float64` precision by doing `jax.config.update("jax_enable_x64", True)` solved my issue perfectly, thank you! Here is the resulting Hessian and Gradient Graph for reference after enabling 64bit precision. 

![image](https://github.com/user-attachments/assets/e4459e79-e9e9-4a4a-958d-e0b8f654c095)

And here I was thinking I would have to make my own custom gradients. Thank you for your help!
